Sources/Kitura/staticFileServer/FileServer.swift

```diff
@@ -156,6 +156,85 @@ extension StaticFileServer {
             }
             response.statusCode = .OK
         }
+        
+        func streamFile(_ filePath: String, request: RouterRequest, response: RouterResponse) {
+            let fileManager = FileManager()
+            var isDirectory = ObjCBool(false)
+            
+            if fileManager.fileExists(atPath: filePath, isDirectory: &isDirectory) {
+                #if os(Linux)
+                    let isDirectoryBool = isDirectory
+                #else
+                    let isDirectoryBool = isDirectory.boolValue
+                #endif
+                
+                if !isDirectoryBool {
+                    streamNonDirectoryFile(filePath, request: request, response: response)
+                    return
+                }
+            }
+        }
+        
+        @discardableResult
+        private func streamNonDirectoryFile(_ filePath: String, request: RouterRequest, response: RouterResponse) -> Bool {
+            if  !isValidFilePath(filePath) {
+                return false
+            }
+            
+            do {
+                let fileAttributes = try FileManager().attributesOfItem(atPath: filePath)
+                responseHeadersSetter?.setCustomResponseHeaders(response: response,
+                                                                filePath: filePath,
+                                                                fileAttributes: fileAttributes)
+                
+                let rangeInfo = request.headers["Range"]?.components(separatedBy: "=")
+                
+                guard let rangeUnit = rangeInfo?.first,
+                    rangeUnit == "bytes",
+                    var byteRanges = rangeInfo?.last?.components(separatedBy: "-"),
+                    byteRanges.count == 2 else {
+                    response.statusCode = .badRequest
+                    return false
+                }
+                
+                guard let streamData = try? Data(contentsOf: URL(fileURLWithPath: filePath), options: [.mappedIfSafe]) else {
```

Comment:
I can change that, however I thought that passing .mappedIfSafe was enough here. Will update the PR